fearlessTests/Modules/Staking/BondMoreConfirm/BondMoreConfirmTests.swift

```diff
@@ -0,0 +1,178 @@
+        let repository: CoreDataRepository<StashItem, CDStashItem> =
+            storageFacade.createRepository()
+
+        let operationQueue = OperationQueue()
+        let saveStashItemOperation = repository.saveOperation({ [stashItem] }, { [] })
+        operationQueue.addOperations([saveStashItemOperation], waitUntilFinished: true)
+
+        let substrateProviderFactory = SubstrateDataProviderFactory(
+            facade: storageFacade,
+            operationManager: operationManager
+        )
+
+        let runtimeCodingService = try RuntimeCodingServiceStub.createWestendService()
+
+        let accountRepository: CoreDataRepository<AccountItem, CDAccountItem> =
+            UserDataStorageTestFacade().createRepository()
+        let anyAccountRepository = AnyDataProviderRepository(accountRepository)
+
+        // save controller
+        let controllerItem = settings.selectedAccount!
+        let saveControllerOperation = anyAccountRepository.saveOperation({ [controllerItem] }, { [] })
+        operationQueue.addOperations([saveControllerOperation], waitUntilFinished: true)
+
+        let extrinsicServiceFactory = ExtrinsicServiceFactoryStub(
+            extrinsicService: ExtrinsicServiceStub.dummy(),
+            signingWraper: try DummySigner(cryptoType: cryptoType)
+        )
+
+        let interactor = StakingBondMoreConfirmationInteractor(
+            settings: settings,
+            singleValueProviderFactory: singleValueProviderFactory,
+            substrateProviderFactory: substrateProviderFactory,
+            accountRepository: anyAccountRepository,
+            extrinsicServiceFactory: extrinsicServiceFactory,
+            feeProxy: ExtrinsicFeeProxy(),
+            runtimeService: runtimeCodingService,
+            operationManager: operationManager,
+            chain: chain,
+            assetId: assetId
+        )
+
+        let balanceViewModelFactory = BalanceViewModelFactory(
+            walletPrimitiveFactory: primitiveFactory,
+            selectedAddressType: chain.addressType,
+            limit: StakingConstants.maxAmount
+        )
+
+        let confirmViewModelFactory = StakingBondMoreConfirmViewModelFactory(asset: asset)
+
+        let presenter = StakingBondMoreConfirmationPresenter(
+            interactor: interactor,
+            wireframe: wireframe,
+            inputAmount: inputAmount,
+            confirmViewModelFactory: confirmViewModelFactory,
+            balanceViewModelFactory: balanceViewModelFactory,
+            dataValidatingFactory: StakingDataValidatingFactory(presentable: wireframe),
+            chain: chain
+        )
+
+        presenter.view = view
+        interactor.presenter = presenter
+
+        // when
+
+        let feeExpectation = XCTestExpectation()
+        let assetExpectation = XCTestExpectation()
+        let confirmViewModelExpectation = XCTestExpectation()
+
+        stub(view) { stub in
```

Comment:
setupPresenter method has "testing" side-effect? 🧐

Reply:
Yep, will think how to improve. Thanks.